pensjon-brevbaker/src/main/kotlin/no/nav/pensjon/brev/latex/LaTeXCompilerService.kt

```diff
@@ -3,30 +3,47 @@ package no.nav.pensjon.brev.latex
 import io.ktor.client.*
 import io.ktor.client.call.*
 import io.ktor.client.engine.cio.*
+import io.ktor.client.plugins.*
 import io.ktor.client.plugins.contentnegotiation.*
 import io.ktor.client.request.*
 import io.ktor.http.*
 import io.ktor.serialization.jackson.*
+import no.nav.pensjon.brev.template.render.RenderedLatexLetter
 
 data class PdfCompilationInput(val files: Map<String, String>)
 data class PDFCompilationOutput(val base64PDF: String)
 
+class LatexCompileException(msg: String, cause: Throwable? = null) : Exception(msg, cause)
+
 class LaTeXCompilerService(private val pdfByggerUrl: String) {
     private val httpClient = HttpClient(CIO) {
         install(ContentNegotiation) {
             jackson()
         }
 
+        HttpResponseValidator {
+            validateResponse { response ->
+                when (response.status) {
+                    HttpStatusCode.BadRequest -> {
+                        throw LatexCompileException("Rendered latex is invalid, couldn't compile pdf: ${response.body<String>()}")
+                    }
+                    HttpStatusCode.InternalServerError -> {
+                        throw LatexCompileException("Couldn't compile latex to pdf due to server error: ${response.body<String>()}")
+                    }
+                }
+            }
+        }
+
         engine {
             requestTimeout = 120_000
         }
     }
 
-    suspend fun producePDF(compilationInput: PdfCompilationInput, callId: String?): PDFCompilationOutput =
+    suspend fun producePDF(latexLetter: RenderedLatexLetter, callId: String?): PDFCompilationOutput =
```

Comment:
GeneratedLatexLetter? Rendered så tenker jeg PDF.

Reply:
Det er jo output fra PensjonLatexRenderer, så for meg virker RenderedLatexLetter mer intuitivt. Latex != PDF